$OpenBSD: patch-source_build_include_polymer_h,v 1.1 2021/12/08 05:46:25 jsg Exp $

backport 864691defbc55453071050b5cc8408079a8f3199 to fix llvm 13 build
engine: use inthash_t for _prprograminfo lookups instead of a 300MB array

Index: source/build/include/polymer.h
--- source/build/include/polymer.h.orig
+++ source/build/include/polymer.h
@@ -441,7 +441,7 @@ static void         polymer_getscratchmaterial(_prmate
 static _prbucket*   polymer_getbuildmaterial(_prmaterial* material, int16_t tilenum, char pal, int8_t shade, int8_t vis, int32_t cmeth);
 static int32_t      polymer_bindmaterial(const _prmaterial *material, const int16_t* lights, int lightcount);
 static void         polymer_unbindmaterial(int32_t programbits);
-static void         polymer_compileprogram(int32_t programbits);
+static _prprograminfo *polymer_compileprogram(int32_t programbits);
 // LIGHTS
 static void         polymer_removelight(int16_t lighti);
 static void         polymer_updatelights(void);
